wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/CommandScheduler.java

```diff
@@ -86,7 +88,10 @@ public static synchronized CommandScheduler getInstance() {
   // scheduled/canceled during run
   private boolean m_inRunLoop;
   private final Set<Command> m_toSchedule = new LinkedHashSet<>();
-  private final List<Command> m_toCancel = new ArrayList<>();
+  private final Deque<Command> m_toCancel = new ArrayDeque<>();
+
+  private boolean m_deferCancel;
```

Comment:
Why not reuse the `m_inRunLoop` variable (probably rename it)?
Also, I think we'd also want to do something similar for scheduling.

Reply:
Reusing `m_inRunLoop` means deferred cancels will land in `m_toCancel`, which doesn't get processed unless `run` is called; we have tests that call `schedule` and `cancel`, but not `run`, which causes tests to fail. It probably won't make a difference, but it's up to us to decide if we want to keep that behavior.

Reply:
I meant process `toCancel` as well: reuse the whole deferring mechanism, without duplicating members. Feel free to extract duplicate code to methods.